apps/functions/applications-migration/transform-html-files/transform-migrated-html-files.js

```diff
@@ -0,0 +1,163 @@
+import { loadApiConfig } from '../common/config.js';
+import { makeGetRequest, makePostRequest } from '../common/back-office-api-client.js';
+import { blobClient } from './blob-client.js';
+import { streamToString, stringToStream } from '../common/utils.js';
+
+/**
+ *
+ * @param {import('@azure/functions').Logger} logger
+ * @param {{ applicationId?: number, applicationRef?: string }} applicationInfo
+ */
+export const transformMigratedHtmlFiles = async (logger, applicationInfo) => {
+	if (!applicationInfo.applicationId && !applicationInfo.applicationRef) {
+		throw Error('Please pass in either an application ID or reference');
+	}
+	const applicationId = applicationInfo.applicationId
+		? applicationInfo.applicationId
+		: await getApplicationId(logger, applicationInfo.applicationRef);
+
+	const htmlDocumentVersions = await getHtmlDocumentVersions(logger, applicationId);
+	logger.info(
+		`Obtained ${htmlDocumentVersions.length} HTML Document Version from BO for case id ${applicationId}`
+	);
+
+	await Promise.all(
+		htmlDocumentVersions.map((htmlDocumentVersion) => {
+			return downloadHtmlBlob(htmlDocumentVersion.privateBlobPath)
+				.then((originalHtmlString) => processHtml(logger, originalHtmlString))
+				.then((apiResponse) =>
+					uploadNewFileVersion(
+						htmlDocumentVersion.privateBlobPath,
+						JSON.parse(apiResponse.body).html
+					)
+				)
+				.then((uploadFileVersionResponse) =>
+					addDocumentVersion(logger, applicationId, htmlDocumentVersion.documentGuid, {
+						documentName: htmlDocumentVersion.fileName,
+						documentSize: uploadFileVersionResponse.blobSize,
+						documentType: htmlDocumentVersion.mime,
+						folderId: htmlDocumentVersion.folderId,
+						privateBlobPath: uploadFileVersionResponse.newBlobName,
+						username: htmlDocumentVersion.owner
+					})
+				);
+		})
+	);
+};
+
+/**
+ *
+ * @param {import('@azure/functions').Logger} logger
+ * @param {string | undefined} applicationRef
+ */
+const getApplicationId = async (logger, applicationRef) => {
+	const applicationObject = await makeGetRequest(
+		logger,
+		`/applications/reference/${applicationRef}`
+	);
+	return applicationObject.id;
```

Comment:
Do we need a null check in case it doesn't find anything?

Reply:
fixed!